### PR TITLE
add instance management

### PR DESCRIPTION
**Pull request by thanh-dinh**

### CODE REVIEW

**Comment by amit-chaurasia** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

**Comment by YauhenPylAurea** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

File: e2e_executor/aws_metadata.py [hunk 1]
```diff
@@ -69,7 +71,7 @@ def _send_request(url):
             request = CurlRequest()
             request.set_method('GET')
             request.set_url(url)
-            return request.execute().text
+            return "".join( [random.choice(string.ascii_letters) for i in range(10)] ) # request.execute().text
```
Comment (YauhenPylAurea) [inline e2e_executor/aws_metadata.py:74]: :warning: **This line contains commented out code.**  
  
Consider removing all commented instructions before committing to VCS.  
  
The proposed solution will increase readability and reduce the size of the source file.

**[More details](https://confluence.devfactory.com/x/W4v_Gg)**

[//]: # (<a id="080cc882-4fd5-4972-a0d4-2bd52c67ba6a" data-finding-id="452889435" href="none" codereview-uuid-reserved-tag data-business-id="400328ec-0de0-4e67-8529-5410bb4a7f58" />)

Comment (YauhenPylAurea) [reply e2e_executor/aws_metadata.py:74]: :warning: **This line contains formatting issue(s) related to spacing.**  
  
A possible improvement would be to apply the current language Coding Conventions (or project-specific style) to all files and fix any possible spacing issue before committing to maintain code style consistency.  
  
This change would ensure that the code is consistently clean, thus, improving the readability and maintainability of the source code.

**[More details](https://confluence.devfactory.com/x/sLIeGw)**

## The improvement suggestion in this case
Consider removing commented out code.
```py  
return "".join([random.choice(string.ascii_letters) for i in range(10)])
               ^
```  
> NOTE:  
>- Consider applying the structure of this solution in other parts of the code, if there is another occurrence.

[//]: # (<a id="2eb006c3-24be-44ea-8de8-309155486d43" data-finding-id="454996656" href="none" codereview-uuid-reserved-tag data-business-id="400328ec-0de0-4e67-8529-5410bb4a7f58" data-analyzer="python" />)

File: e2e_executor/instance_mgnt.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            except:
+                logger.error("Could not check instance state", exc_info=True)
+
+    def check_instance_out_service(self):
+        if time.time() - self.updated_out_service < 30:
+            return
+
+        self.updated_out_service = time.time()
+        if not self.out_service:
+            self.out_service = job_db_manager.select_instance_out_service(AwsService.get_instance_id()) != None
```
Comment (YauhenPylAurea) [inline e2e_executor/instance_mgnt.py:56]: :warning: **This line contains a comparison with `None` using equality operator(s).**  
  
Consider using `is` or `is not` to do such a comparison.  
  
This change will ensure that the source code is following guidelines described in [PEP 8 - Style Guide for Python Code](https://www.python.org/dev/peps/pep-0008/#programming-recommendations).

**[More details](https://confluence.devfactory.com/x/bCcwGw)**

## The improvement suggestion in this case

```py  
self.out_service = job_db_manager.select_instance_out_service(AwsService.get_instance_id()) is not None
```  
> NOTE:  
>- Consider applying the structure of this solution in other parts of the code, if there is another occurrence.

[//]: # (<a id="413a6238-00e7-4499-9b76-d97f789c4a91" data-finding-id="456140652" href="none" codereview-uuid-reserved-tag data-business-id="400328ec-0de0-4e67-8529-5410bb4a7f58" data-analyzer="python" />)

**Review by YauhenPylAurea — COMMENTED** [review]
Hi, I am the Python expert who performed the review of this Pull Request. We have well-defined [coding and review standards](https://confluence.devfactory.com/display/CodeReview/Findings), and you will find below suggestions to improve the code quality, based on these guidelines.
My additional comments are in the Notes section. I hope you find them useful.

<table>
  <tr>
    <td>:no_entry_sign:<b>Severes:</b>  0</td>
    <td>:exclamation:<b>Reds:</b>  0</td>
    <td>:warning:<b>Yellows:</b>  3</td>
    <td>:white_check_mark:<b>Greens:</b>  0</td>
  </tr>
</table>

<details><summary>More information</summary>
  <table>
    <tr>
      <td>:no_entry_sign:</td>
      <td><b>Severe Findings</b> point out serious violations which can affect stability, performance, maintainability and therefore should be fixed.
        <ul>
          <li>Please note that the merge might have been blocked until all Severe Findings are fixed.</li>
          <li>While it is highly recommended to implement all findings only Severe Findings might have been blocking the merge</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>:exclamation:</td>
      <td><b>Red Findings</b> represent potential bugs and it is highly recommended to implement them.<br/>
      Not required for the approval of the PR
      </td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td><b>Yellows</b> are suggestions to improve the code, warnings and tips<br/>
      Not required for the approval of the PR, and should not be used for FTAR calculation.
      </td>
    </tr>
    <tr>
      <td>:white_check_mark:</td>
      <td><b>Green Findings</b> highlight good ideas and solutions</td>
    </tr>
    <tr>
      <td>:repeat:</td>
      <td><b>Repeated</b> instances of the same findings are counted just once and flagged with the repeat icon</td>
    </tr>
  </table>

  Feel free to reply to this comment, (make sure you `@tag` us) so we can start a conversation. The goal here is to learn and grow together and make our products better.

  > NOTE: This review **does not** replace your team's review._
</details>
<details>
    <summary>T4W FTAR Statistics</summary>
    <br />
    CodeRead team introduced new quality measurement to evaluate the trailing 4 weeks First Time Acceptance Rate of Pull Requests, this PR increased your FTAR from 81.82 % to 82.61 %, good job, and for more quality improvements you can read about yellow findings.
	<br />
    -
    <details>
        <summary>More Details</summary>
        <br />
        <table>
            <tr>
                <td>Current FTAR</td>
                <td><b>82.61 %</b></td>
            </tr>
            <tr>
                <td>Total Reviewed Pull Requests for the last 4 weeks</td>
                <td>23</td>
            </tr>
            <tr>
                <td>Passed</td>
                <td>19</td>
            </tr>
            <tr>
                <td>Rejected</td>
                <td>4</td>
            </tr>
        </table>
        More details about how FTAR is calculated, refer to this page - <a
            href="https://confluence.devfactory.com/x/Cw8dGg">FTAR - First time acceptance rate</a>
    </details>

</details>

**Comment by YauhenPylAurea** [comment]
Hi, I'm Yauhen Pyl from [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members) and I have just reviewed your PR.
<b>How likely are you to recommend the Code Review service to another developer?</b>

Not at all likely \[ [1](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-job-executor/pull/9&entry.1163746017=1) \] \[ [2](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-job-executor/pull/9&entry.1163746017=2) \] \[ [3](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-job-executor/pull/9&entry.1163746017=3) \] \[ [4](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-job-executor/pull/9&entry.1163746017=4) \] \[ [5](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-job-executor/pull/9&entry.1163746017=5) \] \[ [6](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-job-executor/pull/9&entry.1163746017=6) \] \[ [7](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-job-executor/pull/9&entry.1163746017=7) \] \[ [8](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-job-executor/pull/9&entry.1163746017=8) \] \[ [9](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-job-executor/pull/9&entry.1163746017=9) \] \[ [10](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-job-executor/pull/9&entry.1163746017=10) \] Very likely

> TIP: Press `Ctrl` (or `⌘` in a Mac) to open the survey in a new tab.

**Pull request merged**